enclave-tls/src/tls_wrappers/api/tls_wrapper_verify_certificate_extension.c

```diff
@@ -16,22 +16,24 @@ tls_wrapper_err_t tls_wrapper_verify_certificate_extension(tls_wrapper_ctx_t *tl
 {
 	ETLS_DEBUG("tls_wrapper_verify_certificate_extension() called with evidence type: '%s'\n", evidence->type);
 
-	if (!tls_ctx || !tls_ctx->etls_handle || !tls_ctx->etls_handle->verifier || !tls_ctx->etls_handle->verifier->opts ||
-			!tls_ctx->etls_handle->verifier->opts->verify_evidence)
+	if (!tls_ctx || !tls_ctx->etls_handle)
 		return -TLS_WRAPPER_ERR_INVALID;
 
-	if (strcmp(tls_ctx->etls_handle->verifier->opts->type, evidence->type)) {
-		ETLS_WARN("type doesn't match between verifier '%s' and evidence '%s'\n", tls_ctx->etls_handle->verifier->opts->type, evidence->type);
+	if (tls_ctx->etls_handle->flags & ENCLAVE_TLS_CONF_VERPARA_AUTO) {
```

Comment:
So we don't need to have a ENCLAVE_TLS_CONF_VERPARA_AUTO which simply takes effect on nothing. 

The old codes you deleted here are working as expected: it checks the evidence type and try to select new one if failed.

Reply:
not exactly. old codes will select new type after init failed not verify_evidence failed. in fact the init failure just happens at sgx and non-sgx instance(by ecall failed return err). it can't distinguish between sgx-la and sgx-ecdsa for example.  Anyway once verify_evidence failed, the session will be ended even old codes.
Currently implementation, it can solve empty verifier paratemer and auo-adating and sgx-ecdsa sgx-ecdsa-qve cross-verification issue.

Reply:
The problem here is that init failure is not worth the reason because it will be fixed soon. Assuming it was fixed, there is no so-called "unspecified verifier paratemer“ issue.